packages/devtools-connection/src/utils/event-emitter.js

```diff
@@ -2,138 +2,259 @@
  * License, v. 2.0. If a copy of the MPL was not distributed with this
  * file, You can obtain one at http://mozilla.org/MPL/2.0/. */
 
-/**
- * EventEmitter.
- */
+"use strict";
 
-var EventEmitter = function EventEmitter() {};
-module.exports = EventEmitter;
+const BAD_LISTENER = "The event listener must be a function, or an object that has " +
+                     "`EventEmitter.handler` Symbol.";
+
+const eventListeners = Symbol("EventEmitter/listeners");
+const onceOriginalListener = Symbol("EventEmitter/once-original-listener");
+const handler = Symbol("EventEmitter/event-handler");
+
+const isEventHandler = (listener) =>
+  listener && handler in listener && typeof listener[handler] === "function";
+
+class EventEmitter {
```

Comment:
we should add a comment that this file is adapted from https://searchfox.org/mozilla-central/source/devtools/shared/event-emitter.js lines 0 - 259, and that it should be kept in sync with it. We can also add that logging and devtools.dump.emit functionality has been removed